### ISSUE TITLE
what does stage mean in your DAN.py?

### ISSUE THREAD

**Issue by Edwardmark** [zjjMaiMai/Deep-Alignment-Network-A-convolutional-neural-network-for-robust-face-alignment#12]
I understand stage=1 or 2, but how about stage = 0?How can I train the model stage by stage?

**Comment by Edwardmark** [comment]
Hi,
I first train it with stage=0, then I change to stage = 1, then stage =2, but the error plateau at 5.6%, cannot reach the result in paper as 3.19%, I train with 300W train, and test on 300W public test set?What is your accuracy?Or is there any training tricks you did not mention?
Best,
Edward

**Comment by zjjMaiMai** [comment]
hi,
for your first  question about stage, you can see line 401-405 in DAN.py 
https://github.com/zjjMaiMai/Deep-Alignment-Network-A-convolutional-neural-network-for-robust-face-alignment/blob/a563d8d459b3ca44e46ae467f3c020f88190e5aa/DAN/DAN/DAN.py#L401

performance improve from 5.5% to 5.0% on stage 2. 
how many epoch that you train on stage 1 ?

**Comment by Edwardmark** [comment]
@zjjMaiMai have you change your learning rate?

**Comment by Edwardmark** [comment]
@zjjMaiMai how many epochs have you trained in each stage?I train 50 epochs in stage1 with 300w trainset, and get 5.7% in 300w public test set. But when I train stage2, the loss just cannot decline, and if I donnot decrease lr, the network got unstable with loss up and down. But when I decline the lr to 0.0001, the loss still cannot decrease significantly, remain around 5.6%. Any suggestions?

**Comment by zjjMaiMai** [comment]
@Edwardmark I train 5-15 epochs in stage 1.   You can stop early in stage 1, for reduce overfit.

SEE[https://github.com/MarekKowalski/DeepAlignmentNetwork/issues/12]

**Comment by Edwardmark** [comment]
thanks, I'll try

**Comment by Edwardmark** [comment]
@zjjMaiMai Thanks for you kind reply, could you please tell me how many epochs you have trained in stage 2?

**Comment by zjjMaiMai** [comment]
@Edwardmark 
it depends on dataset. you can try 15-20 for 300W.

**Comment by destiny311** [comment]
@Edwardmark  can you share you test code?thank you!
